### PR TITLE
fix(terminal): derive zsh wrapper ZDOTDIR from %x so non-ASCII WSL logins load .zshrc (#8003)

### LINKED ISSUE #8003: [Bug]: WSL terminal — zsh shell-ready wrapper exports ZDOTDIR as a raw Windows path, so user's .zshrc never loads
### Operating system

Windows

### Orca version

_No response_

## Environment
- Orca 1.4.131 (regression — worked before auto-update from 1.4.97)
- Windows 11 Pro (26100), WSL2 (2.7.3.0), Ubuntu, default shell: zsh + oh-my-zsh
- Repo opened from `\\wsl.localhost\Ubuntu\...`

## Symptom
Opening a WSL terminal in Orca gives a bare default zsh prompt
(`HOSTNAME%`). `~/.zshrc` is never sourced: no theme, no aliases, no PATH setup.
Running zsh normally outside Orca works fine.

## Root cause
The generated shell-ready wrapper at
`%APPDATA%\orca\shell-ready\zsh\.zshenv` ends with:

    export ZDOTDIR='C:\Users\<user>\AppData\Roaming\orca/shell-ready/zsh'

This embeds the Windows-native user-data path *verbatim*. Inside WSL that
path does not exist, so after `.zshenv` runs, zsh cannot find
`$ZDOTDIR/.zshrc` (the wrapper that would source the user's `~/.zshrc`),
and silently loads nothing.

The initial spawn env (`buildWslInteractiveLoginShellCommand`) sets ZDOTDIR
correctly (WSL-visible path), but the *file content* generated by
`getZshShellReadyRcfileContent` / the zshenv template uses the untranslated
Windows path. For WSL sessions it should be translated to `/mnt/c/...`.

## Repro
1. Windows + WSL, user login shell = zsh, any `~/.zshrc` (e.g. oh-my-zsh)
2. Open a repo on `\\wsl.localhost\...` in Orca 1.4.131
3. Open the integrated terminal → bare `HOSTNAME%` prompt, `~/.zshrc` not loaded

## Workaround
In `~/.zshenv`, detect the broken state and source `~/.zshrc` manually:

    if [[ -o interactive && "${ORCA_USER_DATA_PATH:-}" == *\\* ]]; then
      ZDOTDIR="$HOME"; source "$HOME/.zshrc"
    fi

### PR DESCRIPTION
**Pull request by rodboev**
Closes #8003

> **Takeover note:** This PR started as @rodboev's `ORCA_ORIG_ZDOTDIR` Windows-path normalization. It has been reworked (with credit — see the `Co-authored-by` trailer) to a smaller wrapper self-location fix that resolves the deeper root cause, including the non-ASCII trigger the normalization approach did not cover. Rationale for the approach change is in **Trade-offs** below.

## Description

The zsh shell-ready wrapper (`getZshEnvTemplate`) restored `ZDOTDIR` from the **env-imported** `${ZDOTDIR}`. That value is unreliable in exactly the scenario #8003 hits:

- On **Windows + WSL**, the wrappers are generated under the Windows user-data path (a Windows literal is baked into `.zshenv`) but sourced inside WSL via `/mnt/c`. The baked literal doesn't exist in WSL, so the wrapper must recover the runtime dir from `${ZDOTDIR}`.
- For a **non-ASCII Windows username** (e.g. a Korean login `홍길동`), the WSL-visible `ZDOTDIR` path contains UTF-8 bytes in zsh's internal **0x84–0x9D token range**. zsh corrupts environment-imported values containing those bytes while processing startup files. The corrupted `${ZDOTDIR}` failed the self-check, the wrapper fell back to the **unusable baked Windows literal**, and the user's `~/.zshrc` never loaded — the reported bare `HOSTNAME%` prompt with no theme, aliases, or PATH.

**Fix:** derive the wrapper dir from `${${(%):-%x}:h}` — `%x` is zsh's internal script name for the file currently being sourced (`:h` = its dirname). It's regenerated from zsh's correctly-metafied internal state, so it is **not** subject to the env-import corruption. `${ZDOTDIR}` is kept only as a fallback when `%x` expansion yields nothing, and the pre-existing final restore still validates with `-f` before trusting the value.

One template (`getZshEnvTemplate`) feeds local PTYs, the daemon, **and** the Windows→WSL launch path (`buildWslShellArgs` → `ensureShellReadyWrappersAt`), so the single change covers every zsh transport.

## Evidence

zsh 5.9 on Ubuntu (WSL2). Core mechanism, isolated:

```
ENV_METHOD=[/root/.../홍길동(corrupted: �`=길�{`)/shell-ready/zsh]   ENV_HAS_ZSHENV=no
SELF_METHOD=[/root/.../홍길동/shell-ready/zsh]                        SELF_HAS_ZSHENV=yes
```
`${ZDOTDIR}` (env method) is corrupted → self-check fails. `${${(%):-%x}:h}` (self method) stays correct.

End-to-end, using the **real** production-generated wrappers, with the baked literal pointing at a missing path (mirrors the Windows→WSL split):

| Scenario | Before (unfixed) | After (this PR) |
|---|---|---|
| ASCII username, path-view move | `.zshrc` loads ✅ | `.zshrc` loads ✅ |
| **Non-ASCII (홍길동) username** | **`.zshrc` NOT loaded ❌** | **`.zshrc` loads ✅** |

Regression pass against the fixed wrappers (real zsh): ASCII path-move, non-ASCII path-move (both `zsh -i` and `zsh -l -i`), XDG `ZDOTDIR` discovery + `typeset -U path` scoping, and vanilla-no-ZDOTDIR fallback all still behave correctly. Windows unit tests, `oxlint`, and `tsc` pass. A new live-zsh regression test (`loads user .zshrc when the wrapper dir contains a non-ASCII (token-range) path`) runs in Linux CI.

## ELI5

Orca leaves zsh a note ("your real config lives over here") using an address written on an envelope. On Windows→WSL that address is garbled by zsh's mailroom when the username has non-English letters (like Korean) — so zsh gives up and your shell opens with no theme, aliases, or PATH. Instead of reading the garbled envelope, we now ask zsh "which file are you reading right now?" and use the folder it's already sitting in. That answer is never garbled, so your real `~/.zshrc` loads every time.

## Trade-offs / user regressions

**Targeted regression risk: zero.**
- **macOS / Linux / SSH (common case):** the value derived from `%x` is identical to the old `${ZDOTDIR}` value (dirname of the `.zshenv` being sourced = the wrapper dir), so behavior is byte-for-byte unchanged. Verified by live-zsh tests.
- **`%x` availability:** `%x`, the `(%)` prompt flag, and the `:h` modifier are all present since zsh 5.0 — below the supported baseline. If `%x` ever expanded to nothing, we fall back to the old `${ZDOTDIR}` path.
- **Perf:** startup-only, in-process string expansion — no subshell/fork, no added filesystem stat (the simplified fallback drops a redundant `-f`; the existing final restore already validates). Confirmed by an adversarial perf review.
- **SSH/relay** uses a separate template that bakes an always-valid literal (deployed to the same path it's sourced from) and is immune by construction — out of scope, unchanged.

**Why the approach changed from path-normalization:** the original `ORCA_ORIG_ZDOTDIR` normalization operated on a *different* variable (the user's config dir, not the wrapper's `ZDOTDIR` self-restore that #8003 breaks), did **not** fix the non-ASCII corruption (a normalized `/mnt/c/Users/홍길동/…` still corrupts on import), and hardcoded the `/mnt/` automount root (breaking custom-automount users). The `%x` self-location fix is smaller, root-cause, and regression-free.

**Known residual (not a regression):** in the non-ASCII case the *final* interactive `$ZDOTDIR` can still read back corrupted, because zsh re-reads the poisoned env value across startup-file boundaries after our restore runs. This is cosmetic — the user's `~/.zshrc` (prompt, theme, aliases, PATH) loads correctly because those are sourced via the HOME-derived `ORCA_ORIG_ZDOTDIR`, which zsh does **not** corrupt. Before this PR that path was *also* wrong **and** nothing loaded, so this is a strict improvement.

## Adversarial review

Ran adversarial perf and correctness/reliability reviews over the diff and all consumers. No blocker/major findings. The one perf nit (a redundant `stat`) was addressed by simplifying the fallback guard.

### CODE REVIEW

File: src/main/daemon/shell-ready.test.ts [hunk 1]
```diff
@@ -667,6 +667,71 @@ describePosix('daemon shell-ready launch config', () => {
...
+        process.env.ZDOTDIR = previousZdotdir
+      }
+      if (previousHome === undefined) {
+        delete process.env.HOME
+      } else {
+        process.env.HOME = previousHome
+      }
+    }
+  })
+
```
[the remark this answers is not in the document]
Comment (rodboev) [reply src/main/daemon/shell-ready.test.ts]: Thanks. The POSIX edge-case coverage is still present; it just sits below the new Windows rows rather than being removed. The daemon suite still covers trailing-slash wrapper fallback at src/main/daemon/shell-ready.test.ts:735, slash-only fallback at :761, and substring preservation at :788. The provider suite still covers the same POSIX cases at src/main/providers/local-pty-shell-ready.test.ts:711, :734, and :757, and the wrapper .zshenv fallback-chain expectations remain asserted earlier in both files. I rechecked the final branch state before publishing, so there isn't an uncovered regression surface here.
